### DISCUSSION TITLE
Twitch sign in doesn't return username or does it

### DISCUSSION THREAD

**Discussion by Nisthar** [directus/directus#16998] [Archive]
So I followed official guide  https://docs.directus.io/self-hosted/sso-examples.html#twitch for twitch sign in

I can sign in using twitch but the username is always empty. Is there a way to get username after the user signs in?

I found this https://dev.twitch.tv/docs/api/reference#get-users to username details. But it needs the oauth access tokens. how do i get oauth access token (not the directus access token)?

**Comment by aidenfoxx** [comment]
How do you mean the username is always empty? Do you wish to login via username, or set the username somewhere in your application?

There is a PR up to add a hook which returns the OAuth access token and profile information. If that sounds like what you need, I encourage you go show your support! https://github.com/directus/directus/pull/16812 😄

**Reply by Nisthar** [reply]
Yeah i want to get more info like username, firstname from the api. Hooks looks like a good idea to do this. I am using the environment variable `AUTH_TWITCH_SCOPE="openid user:read:email"` But i can't find twitch scope for reading username.

**Reply by aidenfoxx** [reply]
I believe the username should be included by default as `preferred_username`?

**Reply by Nisthar** [reply]
> I believe the username should be included by default as `preferred_username`?

is there a way to make it assign it to username field in the database?

**Reply by aidenfoxx** [reply]
It would be possible once the PR I mentioned is merged.
